wagtail/wagtailsnippets/views/snippets.py

```diff
@@ -211,29 +212,50 @@ def edit(request, app_label, model_name, id):
     })
 
 
-def delete(request, app_label, model_name, id):
+def delete(request, app_label, model_name, id=None):
     model = get_snippet_model_from_url_params(app_label, model_name)
 
     permission = get_permission_name('delete', model)
     if not request.user.has_perm(permission):
         return permission_denied(request)
 
-    instance = get_object_or_404(model, id=id)
+    instances = []
+
+    if id:
+        instance = get_object_or_404(model, id=id)
+        instances.append(instance)
+    else:
+        ids = request.GET.getlist('id')
+        for id in ids:
+            instance = get_object_or_404(model, id=id)
+            instances.append(instance)
```

Comment:
`instances = model.objects.filter(id__in=ids)` would fetch all instances in one query. (The 404 behaviour would be different, but I'm undecided about what the 'correct' 404 behaviour is when working with multiple objects...)